### PR TITLE
Bring platform layer examples

### PR DESCRIPTION
**Pull request by mprse**
Bring platform layer examples +readme files.

### CODE REVIEW

**Comment by mprse** [comment]
@jamesbeyond 

CI fails because `BufferedSerial` class is not available. I have used it instead `UARTSerial` which is now deprecated.

`BufferedSerial` was introduced over one month ago: PR https://github.com/ARMmbed/mbed-os/pull/12207.

I see that CI is doing: `mbed update`. But I'm not sure if this command really updates mbd-os?

File: APIs_Platform/FileHandle/main.cpp [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#include "mbed.h"
+
+static DigitalOut led2(LED2);
+
+// UARTSerial derives from FileHandle
+static UARTSerial device(STDIO_UART_TX, STDIO_UART_RX);
```
Comment (jamesbeyond) [inline APIs_Platform/FileHandle/main.cpp]: I think the UARTSerial will be deprecated soon, can you replace it with the new one:
https://github.com/ARMmbed/mbed-os/blob/master/docs/design-documents/drivers/serial/serial.md

**Comment by jamesbeyond** [comment]
> I see that CI is doing: `mbed update`. But I'm not sure if this command really updates mbd-os?
the local version of `mbed-os.lib` is pointed to the latest master https://github.com/ARMmbed/mbed-os-examples-docs_only/blob/master/mbed-os.lib
`mbed update` should only change mbed-os when we are on release branches

File: APIs_Platform/FileHandle/main.cpp (hunk 1 above)
Comment (mprse) [reply APIs_Platform/FileHandle/main.cpp]: Thanks, changed to `BufferedSerial`.

**Comment by jamesbeyond** [comment]
Hey @mprse, I think I know the problem, it is because your branch is out of date. could rebase your branch to latest master? the CI should just work

**Comment by mprse** [comment]
@jamesbeyond Looks like rebase helped, Thank you!

Now I got a problem with `mbed_app.json`. Locally it works when using `mbed compile`, but here we are using `mbed test` and this might be a problem.

**Comment by jamesbeyond** [comment]
Hi @evedon could you take a look, see if you are happy about these example snippets ?

**Comment by jamesbeyond** [comment]
Maybe one thing I'd like to advise is adding a simple version of license @mprse 
```
/*
 * Copyright (c) 2006-2020 Arm Limited and affiliates.
 * SPDX-License-Identifier: Apache-2.0
 */
```

**Comment by evedon** [comment]
I thought that Martin suggested a 2-line copyright header for code snippets.

**Comment by mprse** [comment]
Added missing license header.

File: APIs_Platform/FileHandle/main.cpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+static DigitalOut led2(LED2);
+
+// BufferedSerial derives from FileHandle
+static BufferedSerial device(STDIO_UART_TX, STDIO_UART_RX);
+
+int main()
+{
+    // Perform device-specific setup
+    device.set_baud(19200);
```
Comment (evedon) [inline APIs_Platform/FileHandle/main.cpp]: Changing the baud rate is not useful for this example.

File: APIs_Platform/FileHandle_sigio/main.cpp [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        }
+        putchar(c);
+        led2 = !led2;
+    }
+}
+
+int main()
+{
+    // BufferedSerial-specific method - all others are from FileHandle base class
+    device.set_baud(19200);
```
Comment (evedon) [inline APIs_Platform/FileHandle_sigio/main.cpp]: Changing the baud rate is not useful for this example.

File: APIs_Platform/ScopedRomWriteLock/main.cpp [hunk 4]
```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2006-2020 Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include "mbed.h"
+
+int main()
+{
+    // Enable execution from RAM while in main
```
Comment (evedon) [inline APIs_Platform/ScopedRomWriteLock/main.cpp]: Comment is for a different API

File: APIs_Platform/FileHandle/README.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+## FileHandle example
+
+The example monitors a serial console device, and every time it outputs a character, send it to the console back and toggle LED2.
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle/README.md]: Fixed as suggested.

File: APIs_Platform/FileHandle/README.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+## FileHandle example
+
+The example monitors a serial console device, and every time it outputs a character, send it to the console back and toggle LED2.
+
+1. Flash the board, and ensure the target's USB is plugged into the PC.
+2. Open serial console (select associated COM port, transmission speed: 19200 bps).
+3. Reset the target.
+4. Every time that device outputs a character, the character should appear in the console and LED2 should be toggled LED2.
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle/README.md]: Fixed as suggested.

File: APIs_Platform/FileHandle_sigio/main.cpp [hunk 7]
```diff
@@ -0,0 +1,39 @@
+// Main thread
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/main.cpp]: Removed.

File: APIs_Platform/FileHandle_sigio/main.cpp [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+static void callback_ex()
+{
+    // always read until data is exhausted - we may not get another
+    // sigio otherwise
+    while (1) {
+        char c;
+        if (device.read(&c, 1) != 1) {
+            break;
+        }
+        putchar(c);
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/main.cpp]: I will check this tomorrow.

File: APIs_Platform/FileHandle_sigio/README.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
+## FileHandle sigio example
+
+The example monitors a serial console device, and every time it outputs a character, send it to the console back and toggle LED2.
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/README.md]: Fixed as suggested.

File: APIs_Platform/FileHandle_sigio/README.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+## FileHandle sigio example
+
+The example monitors a serial console device, and every time it outputs a character, send it to the console back and toggle LED2.
+
+1. Flash the board, and ensure the target's USB is plugged into the PC.
+2. Open serial console (select associated COM port, transmission speed: 19200 bps).
+3. Reset the target.
+4. Every time that device outputs a character, the character should appear in the console and LED2 should be toggled LED2.
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/README.md]: Fixed as suggested.

File: APIs_Platform/Poll/main.cpp [hunk 11]
```diff
@@ -0,0 +1,57 @@
+// Transfer bidirectional data between two ports, acting as a virtual link.
+// poll() is used to monitor both ports for input.
```
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/Poll/main.cpp]: Fixed as suggested.

File: APIs_Platform/FileHandle/main.cpp (hunk 2 above)
Comment (mprse) [reply APIs_Platform/FileHandle/main.cpp]: Removed and updated `README.md`.

File: APIs_Platform/FileHandle_sigio/main.cpp (hunk 3 above)
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/main.cpp]: Removed and updated `README.md`.

File: APIs_Platform/ScopedRomWriteLock/main.cpp (hunk 4 above)
Comment (mprse) [reply APIs_Platform/ScopedRomWriteLock/main.cpp]: Fixed comment.

**Comment by mprse** [comment]
Rebased on the master to resolve conflicts.

File: APIs_Platform/FileHandle_sigio/main.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (mprse) [reply APIs_Platform/FileHandle_sigio/main.cpp]: Confirmed that there was a problem with this example. Fixed as suggested. 
Thank you!

**Comment by jamesbeyond** [comment]
do you happy about the changes @evedon

File: APIs_Platform/ScopedRomWriteLock/README.md [hunk 12]
```diff
@@ -0,0 +1,3 @@
+## ScopedRamExecutionLock example
```
Comment (evedon) [inline APIs_Platform/ScopedRomWriteLock/README.md]: Should be ScopedRomWriteLock

Comment (mprse) [reply APIs_Platform/ScopedRomWriteLock/README.md]: Fixed

**Comment by mprse** [comment]
Rebased on the master.

**Pull request merged**